src/v/config/configuration.cc

```diff
@@ -120,6 +120,18 @@ configuration::configuration()
       "Interval for which all coprocessor offsets are flushed to disk",
       {.visibility = visibility::tunable},
       300000ms) // five minutes
+  , topic_memory_per_partition(
+      *this,
+      "topic_memory_per_partition",
+      "Required memory per partition when creating topics",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      1_MiB)
```

Comment:
This seems quite high, what inside redpanda is eating 1MiB per partition? I'd expect this to be on the order of 1KiB.

Reply:
I'm guessing but also interested in the answer that this is conservative estimate based on existing implementation details (which can be improved later). for example there is state and caches associated with each raft group and in the storage layer every segment index is represented in memory and currently isn't purged under memory pressure.

Reply:
Yeah, this PR is not a statement that I think the memory bounds are OK :-)

This is just based on what currently works empirically.  When the memory usage gets improved, we move the threshold.